task7/task7.1/task7-1.js

```diff
@@ -0,0 +1,72 @@
+let str;
+let words = [];
+let letters = [];
+function task1() {
+    str=getString();
+    words = getAllWords(words, str);
+    letters = getAllRepeatingLetters(letters, words);
+    words = deleteAllRepeatingLetters(words, letters);
+    words = deleteEmptyWords(words);
+    document.getElementById("result1").innerHTML=getResult(words);
+};
+function getResult(words) {
+    let res="";
+    for (let i = 0; i < words.length-1; i++) {
+        res=res.concat(words[i]," ");
+    }
+    res=res.concat(words[words.length-1]);
+    return res;
+}
+function getString() {
+    return document.getElementById("input-Task1").value;
+}
+function getAllWords(words, str) {
+    str = str.toLowerCase();
+    words = str.split(" ");
+    let separators = ["?", "!", ":", ";", ",", "."];
+    for (let i = 0; i < words.length; i++) {
+        for (let j = 0; j < separators.length; j++) {
+            let wordlength = words[i].indexOf(separators[j]);
+            if (wordlength >= 0) {
+                words[i] = words[i].substr(0, wordlength);
+            }
+        }
+    }
```

Comment:
Looks like a split inside a split - you could just iterate through the text and look for all separators with `includes`. And add a check for empty entries